src/main/java/gregtech/api/capability/impl/AbstractRecipeLogic.java

```diff
@@ -230,9 +239,20 @@ protected void updateRecipeProgress() {
                     this.progressTime = Math.max(1, progressTime - 2);
                 }
             }
+        } else {
+            if (isActive)
+                setActive(false);
```

Comment:
Why would these be needed? The active state will already be toggled when `completeRecipe()` is called, and I don't really see a case where you would want to toggle the active state when the recipe is not completed.